eslint-plugin-expensify/prefer-import-module-contents.js

```diff
@@ -18,6 +18,40 @@ function isEverySpecifierImport(specifiers = []) {
     return _.every(specifiers, specifier => specifier.type === 'ImportSpecifier');
 }
 
+/**
+ * Make an exception for propTypes since they are sometimes bundled with modules.
+ *
+ * @param {Array} specifiers
+ * @returns {Boolean}
+ */
+function hasPropTypesSpecifier(specifiers) {
+    return _.some(specifiers, specifier => /proptypes/.test(lodashGet(specifier, 'imported.name', '').toLowerCase()));
+}
+
+/**
+ * @param {String} source
+ * @returns {Boolean}
+ */
+function isHigherOrderComponent(source) {
+    return /with/.test(source.toLowerCase());
+}
+
+/**
+ * @param {String} source
+ * @returns {Boolean}
+ */
+function isContextComponent(source) {
+    return /context|provider/.test(source.toLowerCase());
+}
+
```

Comment:
Can you explain why we're ignoring these types of components?

Reply:
I guess we can re-visit whether we want to do this or not later. But basically there are some really common patterns where an HOC is a default export and it's propTypes or something else is a named export which makes it harder to follow the advice to use `import * as` syntax.

I couldn't work out how to improve this so opted to just make exceptions for things that are basically "internal" libraries like HOCs or Providers. My thinking is that it is OK to treat these like 3rd party libs or Onyx - but mostly I am lazy and didn't want to update too many things at once.